Program.cs

```diff
@@ -75,68 +71,90 @@ public static void UpdatePerson()
                         case 1:
                             Console.WriteLine("Please Enter The New Name!");
                             var newName = Console.ReadLine();
-                            person.setFirstName(newName);
+                            people[id].SetFirstName(newName);
                             break;
                         case 2:
                             Console.WriteLine("Please Enter The New Last Name!");
                             var lastName = Console.ReadLine();
-                            person.setLastName(lastName);
+                            people[id].SetLastName(lastName);
                             break;
                     }
                 }
                 else
+                {
+                    Console.WriteLine("Invalid value");
+                }
+                if (id > people.Count || id < 0 )
                 {
                     Console.WriteLine("There is no Person with this  ID");
                 }
             }
+            else
+            {
+                Console.WriteLine(" to update! ");
+            }
+          
         }
+    
 
-        public static void RemovePerson()
-
+    public static void RemovePerson()
         {
-            Console.Write("Please Enter The Person ID you want to remove : ");
-            var Id = Convert.ToInt32(Console.ReadLine());
-            for (var i = 0; i <= people.Count; i++)
+            ListPerson();
+            if (people.Count!=0)
             {
-                if (Id == people[i].getId())
+                Console.Write("Please Enter The Person ID you want to remove : ");
+                string input = Console.ReadLine();
+                bool isInt = Int32.TryParse(input, out var id);
+                if (isInt)
                 {
-                    people.RemoveAt(i);
+                    people.RemoveAt(id);
                 }
             }
+            else
+            {
+                Console.Write(" to delete! \n");
+            }
         }
 
-        public static void PersonBookExe()
+        public static void RunProgram()
         {
+            PrintMenu();
             var exit = false;
             while (!exit)
             {
-
-                Console.WriteLine("Please Select An Option");
-                var option = Convert.ToInt32(Console.ReadLine());
-                switch (option)
+                Console.WriteLine("\nPlease Select An Option");
```

Comment:
I think you should clean the console and show the menu again every new iteration in while!